### ISSUE TITLE
npm became much less stable

### ISSUE THREAD

**Issue by iliakan** [npm/npm#7251]
NPM became much less stable in recent time.

Yesterday I had a problem with a `passport` module not being installed for `koa-passport` (though specified in a dep).
Also I had a problem with `npm i -g several modules` failing. There was a native build failing, one was depending on another, installed them 1-by-1 then fine (but npm worked like that before).
Today I removed all modules, then `rm -rf ~/.npm; rm -rf /tmp/npm*` and `npm install` now breaks the app (`vinyl` not found, specified in subdeps though).

Things like that happen so often, I kind of dream back a year or two ago.

I see there are really many issues here, so probably I should not add to the heap.

But maybe you can recommend me something? Like stick to npm 1.x? Or 2.0.x?... I need a stable npm install, my projects simply include many modules, no voodoo.

**Comment by othiym23** [comment]
As far as I know, npm's stability has been steadily and significantly improving from 1.4.11 (or so) until now. There have been a couple regressions along the way, but we try to fix those as soon as we know about them.

Can you provide more details about the flakiness you're seeing? Can you provide `package.json` files that repeatedly refuse to install properly?

We're getting close to rolling out the next major version of npm, with a substantially reworked install algorithm, and we can use all the examples of weirdness we can find to ensure that the new version is robust.

**Comment by iliakan** [comment]
Hi @othiym23, thanks for answering.

I'm talking to other node developers and what I see is quite contrary to what you say. People complain about problems regularly.

But maybe I get something wrong. Let's see.
Right now I tried to reinstall global modules, and it failed.

```
root> ls /usr/local/lib/node_modules
npm

/js/javascript-nodejs
root> rm -rf ~/.npm; rm -rf /tmp/npm* ; npm i --verbose -g 6to5 bunyan gulp mocha node-inspector nodemon pm2 webpack
npm info it worked if it ends with ok
npm verb cli [ 'node',
npm verb cli   '/usr/local/bin/npm',
npm verb cli   'i',
npm verb cli   '--verbose',
npm verb cli   '-g',
npm verb cli   '6to5',
npm verb cli   'bunyan',
npm verb cli   'gulp',
npm verb cli   'mocha',
npm verb cli   'node-inspector',
npm verb cli   'nodemon',
npm verb cli   'pm2',
npm verb cli   'webpack' ]
npm info using npm@2.4.1
npm info using node@v0.11.15
npm verb cache add spec 6to5
npm verb cache add spec bunyan
npm verb cache add spec gulp
npm verb cache add spec mocha
npm verb cache add spec node-inspector
npm verb cache add spec nodemon
npm verb cache add spec pm2
npm verb cache add spec webpack
npm verb addNamed 6to5@*
npm verb addNameRange registry:https://registry.npmjs.org/6to5 not in flight; fetching
npm verb addNamed bunyan@*
npm verb addNameRange registry:https://registry.npmjs.org/bunyan not in flight; fetching
npm verb addNamed node-inspector@*
npm verb addNameRange registry:https://registry.npmjs.org/node-inspector not in flight; fetching
npm verb addNamed nodemon@*
npm verb addNameRange registry:https://registry.npmjs.org/nodemon not in flight; fetching
npm verb addNamed webpack@*
npm verb addNameRange registry:https://registry.npmjs.org/webpack not in flight; fetching
npm verb request uri https://registry.npmjs.org/6to5
npm verb request no auth needed
npm info attempt registry request try #1 at 10:35:38
npm verb request id a4a184e965f53fc6
npm http request GET https://registry.npmjs.org/6to5
npm verb request uri https://registry.npmjs.org/bunyan
npm verb request no auth needed
npm info attempt registry request try #1 at 10:35:38
npm http request GET https://registry.npmjs.org/bunyan
npm verb request uri https://registry.npmjs.org/node-inspector
npm verb request no auth needed
npm info attempt registry request try #1 at 10:35:38
npm http request GET https://registry.npmjs.org/node-inspector
npm verb request uri https://registry.npmjs.org/nodemon
npm verb request no auth needed
npm info attempt registry request try #1 at 10:35:38
npm http request GET https://registry.npmjs.org/nodemon
npm verb addNamed pm2@*
npm verb addNameRange registry:https://registry.npmjs.org/pm2 not in flight; fetching
npm verb request uri https://registry.npmjs.org/webpack
npm verb request no auth needed
npm info attempt registry request try #1 at 10:35:38
npm http request GET https://registry.npmjs.org/webpack
npm verb request uri https://registry.npmjs.org/pm2
npm verb request no auth needed
npm info attempt registry request try #1 at 10:35:38
npm http request GET https://registry.npmjs.org/pm2
npm verb addTmpTarball /js/javascript-nodejs/gulp not in flight; adding
npm verb addTmpTarball /js/javascript-nodejs/mocha not in flight; adding
npm verb getCacheStat cache creation not in flight; initializing
npm verb getCacheStat cache creation already in flight; waiting
npm verb addTmpTarball validating metadata from /js/javascript-nodejs/gulp
npm verb tar unpack /js/javascript-nodejs/gulp
npm verb tar unpacking to /tmp/npm-15999-cb47e38d/unpack-4a5dd9a74d26
npm verb addTmpTarball validating metadata from /js/javascript-nodejs/mocha
npm verb tar unpack /js/javascript-nodejs/mocha
npm verb tar unpacking to /tmp/npm-15999-cb47e38d/unpack-3e2623d89e90
npm verb gentlyRm vacuuming /tmp/npm-15999-cb47e38d/unpack-4a5dd9a74d26
npm verb gentlyRm vacuuming /tmp/npm-15999-cb47e38d/unpack-3e2623d89e90
npm ERR! not a package /js/javascript-nodejs/mocha
npm ERR! addLocal Could not install /js/javascript-nodejs/mocha
npm ERR! not a package /js/javascript-nodejs/gulp
npm ERR! addLocal Could not install /js/javascript-nodejs/gulp
npm http 200 https://registry.npmjs.org/6to5
npm verb get saving 6to5 to /Users/iliakan/.npm/registry.npmjs.org/6to5/.cache.json
npm verb addNamed 6to5@3.3.0
npm verb addRemoteTarball https://registry.npmjs.org/6to5/-/6to5-3.3.0.tgz not in flight; adding
npm verb addRemoteTarball [ 'https://registry.npmjs.org/6to5/-/6to5-3.3.0.tgz',
npm verb addRemoteTarball   '71e1f5e2c21d81c821c66718bbe4303f5e17030c' ]
npm info retry fetch attempt 1 at 10:35:39
npm info attempt registry request try #1 at 10:35:39
npm http fetch GET https://registry.npmjs.org/6to5/-/6to5-3.3.0.tgz
npm http fetch 200 https://registry.npmjs.org/6to5/-/6to5-3.3.0.tgz
npm http 200 https://registry.npmjs.org/pm2
npm verb get saving pm2 to /Users/iliakan/.npm/registry.npmjs.org/pm2/.cache.json
npm http 200 https://registry.npmjs.org/node-inspector
npm verb get saving node-inspector to /Users/iliakan/.npm/registry.npmjs.org/node-inspector/.cache.json
npm verb addNamed node-inspector@0.8.3
npm verb addNamed pm2@0.12.4
npm verb addRemoteTarball https://registry.npmjs.org/node-inspector/-/node-inspector-0.8.3.tgz not in flight; adding
npm verb addRemoteTarball [ 'https://registry.npmjs.org/node-inspector/-/node-inspector-0.8.3.tgz',
npm verb addRemoteTarball   '2eac57771973c6eb49100ef4c9caa6ae0bcd8911' ]
npm verb addRemoteTarball https://registry.npmjs.org/pm2/-/pm2-0.12.4.tgz not in flight; adding
npm verb addRemoteTarball [ 'https://registry.npmjs.org/pm2/-/pm2-0.12.4.tgz',
npm verb addRemoteTarball   '715d015522a59af2459cae5b1cf1b928ccc8aa1c' ]
npm http 200 https://registry.npmjs.org/nodemon
npm verb get saving nodemon to /Users/iliakan/.npm/registry.npmjs.org/nodemon/.cache.json
npm http 200 https://registry.npmjs.org/bunyan
npm verb get saving bunyan to /Users/iliakan/.npm/registry.npmjs.org/bunyan/.cache.json
npm info retry fetch attempt 1 at 10:35:39
npm info attempt registry request try #1 at 10:35:39
npm http fetch GET https://registry.npmjs.org/node-inspector/-/node-inspector-0.8.3.tgz
npm info retry fetch attempt 1 at 10:35:39
npm info attempt registry request try #1 at 10:35:39
npm http fetch GET https://registry.npmjs.org/pm2/-/pm2-0.12.4.tgz
npm verb addNamed nodemon@1.3.5
npm verb addNamed bunyan@1.3.3
npm verb addRemoteTarball https://registry.npmjs.org/nodemon/-/nodemon-1.3.5.tgz not in flight; adding
npm verb addRemoteTarball [ 'https://registry.npmjs.org/nodemon/-/nodemon-1.3.5.tgz',
npm verb addRemoteTarball   '8ec82d02048474c652d8ad3f4d8a23adb13c4be3' ]
npm verb addRemoteTarball https://registry.npmjs.org/bunyan/-/bunyan-1.3.3.tgz not in flight; adding
npm verb addRemoteTarball [ 'https://registry.npmjs.org/bunyan/-/bunyan-1.3.3.tgz',
npm verb addRemoteTarball   'bf4e301c1f0bf888ec678829531f7b5d212e9e81' ]
npm info retry fetch attempt 1 at 10:35:39
npm info attempt registry request try #1 at 10:35:39
npm http fetch GET https://registry.npmjs.org/nodemon/-/nodemon-1.3.5.tgz
npm info retry fetch attempt 1 at 10:35:39
npm info attempt registry request try #1 at 10:35:39
npm http fetch GET https://registry.npmjs.org/bunyan/-/bunyan-1.3.3.tgz
npm http fetch 200 https://registry.npmjs.org/pm2/-/pm2-0.12.4.tgz
npm http fetch 200 https://registry.npmjs.org/node-inspector/-/node-inspector-0.8.3.tgz
npm verb addTmpTarball /tmp/npm-15999-cb47e38d/registry.npmjs.org/6to5/-/6to5-3.3.0.tgz not in flight; adding
npm verb addTmpTarball already have metadata; skipping unpack for 6to5@3.3.0
npm verb afterAdd /Users/iliakan/.npm/6to5/3.3.0/package/package.json not in flight; writing
npm verb afterAdd /Users/iliakan/.npm/6to5/3.3.0/package/package.json written
npm http fetch 200 https://registry.npmjs.org/nodemon/-/nodemon-1.3.5.tgz
npm http fetch 200 https://registry.npmjs.org/bunyan/-/bunyan-1.3.3.tgz
npm verb addTmpTarball /tmp/npm-15999-cb47e38d/registry.npmjs.org/nodemon/-/nodemon-1.3.5.tgz not in flight; adding
npm verb addTmpTarball already have metadata; skipping unpack for nodemon@1.3.5
npm verb afterAdd /Users/iliakan/.npm/nodemon/1.3.5/package/package.json not in flight; writing
npm verb afterAdd /Users/iliakan/.npm/nodemon/1.3.5/package/package.json written
npm verb addTmpTarball /tmp/npm-15999-cb47e38d/registry.npmjs.org/pm2/-/pm2-0.12.4.tgz not in flight; adding
npm verb addTmpTarball already have metadata; skipping unpack for pm2@0.12.4
npm verb afterAdd /Users/iliakan/.npm/pm2/0.12.4/package/package.json not in flight; writing
npm verb afterAdd /Users/iliakan/.npm/pm2/0.12.4/package/package.json written
npm http 200 https://registry.npmjs.org/webpack
npm verb get saving webpack to /Users/iliakan/.npm/registry.npmjs.org/webpack/.cache.json
npm verb addNamed webpack@1.5.3
npm verb addRemoteTarball https://registry.npmjs.org/webpack/-/webpack-1.5.3.tgz not in flight; adding
npm verb addRemoteTarball [ 'https://registry.npmjs.org/webpack/-/webpack-1.5.3.tgz',
npm verb addRemoteTarball   '817a91e1fe70bbaba9400ea49afdfc9a05f4b0c5' ]
npm info retry fetch attempt 1 at 10:35:40
npm info attempt registry request try #1 at 10:35:40
npm http fetch GET https://registry.npmjs.org/webpack/-/webpack-1.5.3.tgz
npm verb addTmpTarball /tmp/npm-15999-cb47e38d/registry.npmjs.org/bunyan/-/bunyan-1.3.3.tgz not in flight; adding
npm verb addTmpTarball already have metadata; skipping unpack for bunyan@1.3.3
npm verb afterAdd /Users/iliakan/.npm/bunyan/1.3.3/package/package.json not in flight; writing
npm http fetch 200 https://registry.npmjs.org/webpack/-/webpack-1.5.3.tgz
npm verb afterAdd /Users/iliakan/.npm/bunyan/1.3.3/package/package.json written
npm verb addTmpTarball /tmp/npm-15999-cb47e38d/registry.npmjs.org/webpack/-/webpack-1.5.3.tgz not in flight; adding
npm verb addTmpTarball already have metadata; skipping unpack for webpack@1.5.3
npm verb afterAdd /Users/iliakan/.npm/webpack/1.5.3/package/package.json not in flight; writing
npm verb afterAdd /Users/iliakan/.npm/webpack/1.5.3/package/package.json written
npm verb addTmpTarball /tmp/npm-15999-cb47e38d/registry.npmjs.org/node-inspector/-/node-inspector-0.8.3.tgz not in flight; adding
npm verb addTmpTarball already have metadata; skipping unpack for node-inspector@0.8.3
npm verb afterAdd /Users/iliakan/.npm/node-inspector/0.8.3/package/package.json not in flight; writing
npm verb afterAdd /Users/iliakan/.npm/node-inspector/0.8.3/package/package.json written
npm verb stack Error: ENOENT, open '/tmp/npm-15999-cb47e38d/unpack-3e2623d89e90/package.json'
npm verb stack     at Error (native)
npm verb cwd /js/javascript-nodejs
npm ERR! Darwin 14.0.0
npm ERR! argv "node" "/usr/local/bin/npm" "i" "--verbose" "-g" "6to5" "bunyan" "gulp" "mocha" "node-inspector" "nodemon" "pm2" "webpack"
npm ERR! node v0.11.15
npm ERR! npm  v2.4.1
npm ERR! path /tmp/npm-15999-cb47e38d/unpack-3e2623d89e90/package.json
npm ERR! code ENOENT
npm ERR! errno -2

npm ERR! enoent ENOENT, open '/tmp/npm-15999-cb47e38d/unpack-3e2623d89e90/package.json'
npm ERR! enoent This is most likely not a problem with npm itself
npm ERR! enoent and is related to npm not being able to find a file.
npm ERR! enoent
npm verb exit [ -2, true ]

npm ERR! Please include the following file with any support request:
npm ERR!     /js/javascript-nodejs/npm-debug.log
```




/js/javascript-nodejs
root> npm i --verbose -g 6to5 bunyan gulp mocha node-inspector nodemon pm2 webpack
npm info it worked if it ends with ok
npm verb cli [ 'node',
npm verb cli   '/usr/local/bin/npm',
npm verb cli   'i',
npm verb cli   '--verbose',
npm verb cli   '-g',
npm verb cli   '6to5',
npm verb cli   'bunyan',
npm verb cli   'gulp',
npm verb cli   'mocha',
npm verb cli   'node-inspector',
npm verb cli   'nodemon',
npm verb cli   'pm2',
npm verb cli   'webpack' ]
npm info using npm@2.4.1
npm info using node@v0.11.15
npm verb cache add spec 6to5
npm verb cache add spec bunyan
npm verb cache add spec gulp
npm verb cache add spec mocha
npm verb cache add spec node-inspector
npm verb cache add spec nodemon
npm verb cache add spec pm2
npm verb cache add spec webpack
npm verb addNamed 6to5@*
npm verb addNameRange registry:https://registry.npmjs.org/6to5 not in flight; fetching
npm verb addNamed bunyan@*
npm verb addNameRange registry:https://registry.npmjs.org/bunyan not in flight; fetching
npm verb addNamed node-inspector@*
npm verb addNameRange registry:https://registry.npmjs.org/node-inspector not in flight; fetching
npm verb addNamed nodemon@*
npm verb addNameRange registry:https://registry.npmjs.org/nodemon not in flight; fetching
npm verb addNamed webpack@*
npm verb addNameRange registry:https://registry.npmjs.org/webpack not in flight; fetching
npm verb request uri https://registry.npmjs.org/6to5
npm verb request no auth needed
npm info attempt registry request try #1 at 10:30:36
npm verb request id 8ba164a3a7a96057
npm http request GET https://registry.npmjs.org/6to5
npm verb request uri https://registry.npmjs.org/bunyan
npm verb request no auth needed
npm info attempt registry request try #1 at 10:30:37
npm http request GET https://registry.npmjs.org/bunyan
npm verb request uri https://registry.npmjs.org/node-inspector
npm verb request no auth needed
npm info attempt registry request try #1 at 10:30:37
npm http request GET https://registry.npmjs.org/node-inspector
npm verb request uri https://registry.npmjs.org/nodemon
npm verb request no auth needed
npm info attempt registry request try #1 at 10:30:37
npm http request GET https://registry.npmjs.org/nodemon
npm verb addNamed pm2@*
npm verb addNameRange registry:https://registry.npmjs.org/pm2 not in flight; fetching
npm verb request uri https://registry.npmjs.org/webpack
npm verb request no auth needed
npm info attempt registry request try #1 at 10:30:37
npm http request GET https://registry.npmjs.org/webpack
npm verb request uri https://registry.npmjs.org/pm2
npm verb request no auth needed
npm info attempt registry request try #1 at 10:30:37
npm http request GET https://registry.npmjs.org/pm2
npm verb addTmpTarball /js/javascript-nodejs/gulp not in flight; adding
npm verb addTmpTarball /js/javascript-nodejs/mocha not in flight; adding
npm verb getCacheStat cache creation not in flight; initializing
npm verb getCacheStat cache creation already in flight; waiting
npm verb addTmpTarball validating metadata from /js/javascript-nodejs/gulp
npm verb tar unpack /js/javascript-nodejs/gulp
npm verb tar unpacking to /tmp/npm-13703-5cf42500/unpack-7635eda29e3e
npm verb addTmpTarball validating metadata from /js/javascript-nodejs/mocha
npm verb tar unpack /js/javascript-nodejs/mocha
npm verb tar unpacking to /tmp/npm-13703-5cf42500/unpack-8aef3b19e7e6
npm verb gentlyRm vacuuming /tmp/npm-13703-5cf42500/unpack-7635eda29e3e
npm verb gentlyRm vacuuming /tmp/npm-13703-5cf42500/unpack-8aef3b19e7e6
npm ERR! not a package /js/javascript-nodejs/mocha
npm ERR! addLocal Could not install /js/javascript-nodejs/mocha
npm ERR! not a package /js/javascript-nodejs/gulp
npm ERR! addLocal Could not install /js/javascript-nodejs/gulp
npm http 200 https://registry.npmjs.org/node-inspector
npm verb get saving node-inspector to /Users/iliakan/.npm/registry.npmjs.org/node-inspector/.cache.json
npm verb addNamed node-inspector@0.8.3
npm verb addRemoteTarball https://registry.npmjs.org/node-inspector/-/node-inspector-0.8.3.tgz not in flight; adding
npm verb addRemoteTarball [ 'https://registry.npmjs.org/node-inspector/-/node-inspector-0.8.3.tgz',
npm verb addRemoteTarball   '2eac57771973c6eb49100ef4c9caa6ae0bcd8911' ]
npm http 200 https://registry.npmjs.org/bunyan
npm verb get saving bunyan to /Users/iliakan/.npm/registry.npmjs.org/bunyan/.cache.json
npm info retry fetch attempt 1 at 10:30:37
npm info attempt registry request try #1 at 10:30:37
npm http fetch GET https://registry.npmjs.org/node-inspector/-/node-inspector-0.8.3.tgz
npm verb addNamed bunyan@1.3.3
npm verb addRemoteTarball https://registry.npmjs.org/bunyan/-/bunyan-1.3.3.tgz not in flight; adding
npm verb addRemoteTarball [ 'https://registry.npmjs.org/bunyan/-/bunyan-1.3.3.tgz',
npm verb addRemoteTarball   'bf4e301c1f0bf888ec678829531f7b5d212e9e81' ]
npm http 200 https://registry.npmjs.org/nodemon
npm verb get saving nodemon to /Users/iliakan/.npm/registry.npmjs.org/nodemon/.cache.json
npm http 200 https://registry.npmjs.org/pm2
npm verb get saving pm2 to /Users/iliakan/.npm/registry.npmjs.org/pm2/.cache.json
npm http fetch 200 https://registry.npmjs.org/node-inspector/-/node-inspector-0.8.3.tgz
npm info retry fetch attempt 1 at 10:30:37
npm info attempt registry request try #1 at 10:30:37
npm http fetch GET https://registry.npmjs.org/bunyan/-/bunyan-1.3.3.tgz
npm verb addNamed nodemon@1.3.5
npm verb addNamed pm2@0.12.4
npm verb addRemoteTarball https://registry.npmjs.org/nodemon/-/nodemon-1.3.5.tgz not in flight; adding
npm verb addRemoteTarball [ 'https://registry.npmjs.org/nodemon/-/nodemon-1.3.5.tgz',
npm verb addRemoteTarball   '8ec82d02048474c652d8ad3f4d8a23adb13c4be3' ]
npm verb addRemoteTarball https://registry.npmjs.org/pm2/-/pm2-0.12.4.tgz not in flight; adding
npm verb addRemoteTarball [ 'https://registry.npmjs.org/pm2/-/pm2-0.12.4.tgz',
npm verb addRemoteTarball   '715d015522a59af2459cae5b1cf1b928ccc8aa1c' ]
npm http 200 https://registry.npmjs.org/6to5
npm verb get saving 6to5 to /Users/iliakan/.npm/registry.npmjs.org/6to5/.cache.json
npm http fetch 200 https://registry.npmjs.org/bunyan/-/bunyan-1.3.3.tgz
npm info retry fetch attempt 1 at 10:30:37
npm info attempt registry request try #1 at 10:30:37
npm http fetch GET https://registry.npmjs.org/nodemon/-/nodemon-1.3.5.tgz
npm info retry fetch attempt 1 at 10:30:37
npm info attempt registry request try #1 at 10:30:37
npm http fetch GET https://registry.npmjs.org/pm2/-/pm2-0.12.4.tgz
npm verb addNamed 6to5@3.3.0
npm verb addRemoteTarball https://registry.npmjs.org/6to5/-/6to5-3.3.0.tgz not in flight; adding
npm verb addRemoteTarball [ 'https://registry.npmjs.org/6to5/-/6to5-3.3.0.tgz',
npm verb addRemoteTarball   '71e1f5e2c21d81c821c66718bbe4303f5e17030c' ]
npm info retry fetch attempt 1 at 10:30:37
npm info attempt registry request try #1 at 10:30:37
npm http fetch GET https://registry.npmjs.org/6to5/-/6to5-3.3.0.tgz
npm http fetch 200 https://registry.npmjs.org/pm2/-/pm2-0.12.4.tgz
npm http fetch 200 https://registry.npmjs.org/nodemon/-/nodemon-1.3.5.tgz
npm http 200 https://registry.npmjs.org/webpack
npm verb get saving webpack to /Users/iliakan/.npm/registry.npmjs.org/webpack/.cache.json
npm http fetch 200 https://registry.npmjs.org/6to5/-/6to5-3.3.0.tgz
npm verb addNamed webpack@1.5.3
npm verb addRemoteTarball https://registry.npmjs.org/webpack/-/webpack-1.5.3.tgz not in flight; adding
npm verb addRemoteTarball [ 'https://registry.npmjs.org/webpack/-/webpack-1.5.3.tgz',
npm verb addRemoteTarball   '817a91e1fe70bbaba9400ea49afdfc9a05f4b0c5' ]
npm info retry fetch attempt 1 at 10:30:37
npm info attempt registry request try #1 at 10:30:37
npm http fetch GET https://registry.npmjs.org/webpack/-/webpack-1.5.3.tgz
npm verb addTmpTarball /tmp/npm-13703-5cf42500/registry.npmjs.org/nodemon/-/nodemon-1.3.5.tgz not in flight; adding
npm verb addTmpTarball already have metadata; skipping unpack for nodemon@1.3.5
npm verb afterAdd /Users/iliakan/.npm/nodemon/1.3.5/package/package.json not in flight; writing
npm verb afterAdd /Users/iliakan/.npm/nodemon/1.3.5/package/package.json written
npm http fetch 200 https://registry.npmjs.org/webpack/-/webpack-1.5.3.tgz
npm verb addTmpTarball /tmp/npm-13703-5cf42500/registry.npmjs.org/bunyan/-/bunyan-1.3.3.tgz not in flight; adding
npm verb addTmpTarball already have metadata; skipping unpack for bunyan@1.3.3
npm verb afterAdd /Users/iliakan/.npm/bunyan/1.3.3/package/package.json not in flight; writing
npm verb afterAdd /Users/iliakan/.npm/bunyan/1.3.3/package/package.json written
npm verb addTmpTarball /tmp/npm-13703-5cf42500/registry.npmjs.org/6to5/-/6to5-3.3.0.tgz not in flight; adding
npm verb addTmpTarball already have metadata; skipping unpack for 6to5@3.3.0
npm verb addTmpTarball /tmp/npm-13703-5cf42500/registry.npmjs.org/webpack/-/webpack-1.5.3.tgz not in flight; adding
npm verb addTmpTarball already have metadata; skipping unpack for webpack@1.5.3
npm verb afterAdd /Users/iliakan/.npm/6to5/3.3.0/package/package.json not in flight; writing
npm verb afterAdd /Users/iliakan/.npm/6to5/3.3.0/package/package.json written
npm verb afterAdd /Users/iliakan/.npm/webpack/1.5.3/package/package.json not in flight; writing
npm verb afterAdd /Users/iliakan/.npm/webpack/1.5.3/package/package.json written
npm verb addTmpTarball /tmp/npm-13703-5cf42500/registry.npmjs.org/pm2/-/pm2-0.12.4.tgz not in flight; adding
npm verb addTmpTarball already have metadata; skipping unpack for pm2@0.12.4
npm verb afterAdd /Users/iliakan/.npm/pm2/0.12.4/package/package.json not in flight; writing
npm verb afterAdd /Users/iliakan/.npm/pm2/0.12.4/package/package.json written
npm verb addTmpTarball /tmp/npm-13703-5cf42500/registry.npmjs.org/node-inspector/-/node-inspector-0.8.3.tgz not in flight; adding
npm verb addTmpTarball already have metadata; skipping unpack for node-inspector@0.8.3
npm verb afterAdd /Users/iliakan/.npm/node-inspector/0.8.3/package/package.json not in flight; writing
npm verb afterAdd /Users/iliakan/.npm/node-inspector/0.8.3/package/package.json written
npm verb stack Error: ENOENT, open '/tmp/npm-13703-5cf42500/unpack-8aef3b19e7e6/package.json'
npm verb stack     at Error (native)
npm verb cwd /js/javascript-nodejs
npm ERR! Darwin 14.0.0
npm ERR! argv "node" "/usr/local/bin/npm" "i" "--verbose" "-g" "6to5" "bunyan" "gulp" "mocha" "node-inspector" "nodemon" "pm2" "webpack"
npm ERR! node v0.11.15
npm ERR! npm  v2.4.1
npm ERR! path /tmp/npm-13703-5cf42500/unpack-8aef3b19e7e6/package.json
npm ERR! code ENOENT
npm ERR! errno -2

npm ERR! enoent ENOENT, open '/tmp/npm-13703-5cf42500/unpack-8aef3b19e7e6/package.json'
npm ERR! enoent This is most likely not a problem with npm itself
npm ERR! enoent and is related to npm not being able to find a file.
npm ERR! enoent
npm verb exit [ -2, true ]

npm ERR! Please include the following file with any support request:
npm ERR!     /js/javascript-nodejs/npm-debug.log

**Comment by iliakan** [comment]
P.S. That was on my Macbook. A totally fresh install on Debian 7.7. passes.

Yeah, there are artifacts like:
```
gyp WARN EACCES user "root" does not have permission to access the dev dir "/root/.node-gyp/0.11.15"
gyp WARN EACCES attempting to reinstall using temporary dev dir "/usr/local/lib/node_modules/node-inspector/node_modules/v8-profiler/.node-gyp"
```

...But it actually works. Did I forget to clean up something?  (besides /tmp/npm* and ~/.npm).

**Comment by othiym23** [comment]
While this is no means conclusive, I just ran the same commands as you, adjusted for Windows, and had no issues. I was using Windows 8.1, Node 0.10.35, and `npm@2.5.0`. The error in the `npm-debug.log` you included looks like the consequence of a race condition, which would be consistent with it being only having errors some of the time. I'll have to take a closer look at `npm-debug.log` and see if I can figure out what's going on, because none of those packages has a particularly complicated installer.

> Did I forget to clean up something?

I don't think so. Because the temporary directories use randomly-generated suffixes for their names, even deleting them shouldn't be required. `npm cache clean` should be sufficient to reset the state of npm's world (although, of course, `node-gyp` caches headers and the like in `$HOME/.node-gyp`, but that shouldn't make any difference).

If you have the time to reduce the install failure you're seeing on OS X down to the smallest subset of packages that can even semi-reliably force this error, I can take it from there. It's possible that there's something going on with Node 0.11.15 and npm (although I doubt it; the error looks like it has to do with a file not getting extracted from the package tarball, or perhaps the installer trying to read a `package.json` before the tarball has finished unpacking -- neither of which should be affected by the Node version), and of course it's possible there's yet another race condition lurking in the caching code.

**Comment by iliakan** [comment]
And then I tried to install local modules for https://github.com/iliakan/javascript-nodejs, after some time the process finished, but something's wrong with the deps.

First time it passed, but `koa-passport` did not have `passport` internal dependancy ( in package.json, but not installed).

On the second pass I see this:

, I see this error:
```
npm ERR! Darwin 14.0.0
npm ERR! argv "node" "/usr/local/bin/npm" "i"
npm ERR! node v0.11.15
npm ERR! npm  v2.4.1
npm ERR! code ELIFECYCLE

npm ERR! pngquant-bin@2.0.3 postinstall: `node lib/install.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the pngquant-bin@2.0.3 postinstall script 'node lib/install.js'.
npm ERR! This is most likely a problem with the pngquant-bin package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node lib/install.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls pngquant-bin
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /js/javascript-nodejs/npm-debug.log
```

And also, passport, once again:
```
/js/javascript-nodejs
root> ls node_modules/koa-passport/node_modules/
co
```

See, no `passport` ^^^

The log: https://gist.github.com/fac1b7d0cd8b922662f9

**Comment by iliakan** [comment]
P.S. There are 2 problems above:
1) With passport dep
2) With pngquant-bin, I can install it normally with `npm i pngquant-bin`, but not with a `npm i` (from package.json).

**Comment by iliakan** [comment]
There's indeed something phantom about these bugs :/

**Comment by iliakan** [comment]
It's strange that the issue is marked as support.
That's clearly a bug in NPM, it doesn't install subdep.

It was working correctly some time ago.

**Comment by KenanY** [comment]
@iliakan Hello, I just cloned https://github.com/iliakan/javascript-nodejs to try and replicate this. I removed [line 81](https://github.com/iliakan/javascript-nodejs/blob/f9cc60654eb6967f6784e16c69aa431f65de5054/package.json#L81) from the `package.json` since that's your workaround, right? Ran `npm install`, and looks like `npm`'s deduping puts `passport` in your root `node_modules` (so no need for it in `package.json`?):

``` sh
$ npm ls passport
javascript-nodejs@0.0.1 /home/kenan/Code/javascript-nodejs
├── passport@0.2.1 
└─┬ passport-github@0.1.5
  └─┬ passport-oauth@0.1.15
    └── passport@0.1.18 
```

If I go into the same directory as `koa-passport`'s `index.js` (which is where it is doing `require('passport')`), and do a `require.resolve` for `passport`, I correctly receive that root `passport`:

``` sh
$ cd node_modules/koa-passport/lib/
$ node
> require.resolve('passport')
'/home/kenan/Code/javascript-nodejs/node_modules/passport/lib/index.js'
```

So `require('passport').framework` is in fact a function:

``` sh
$ node
> require('passport').framework
[Function]
```

I'm on `npm@2.5.1`. I gotta go offline for a bit now so I haven't tried running your test suite yet.
